### PR TITLE
Add image resource field

### PR DESCRIPTION
**Pull request by AlexandreSi**
Use it for particle emitter "Change image" action

### CODE REVIEW

File: Extensions/ParticleSystem/ExtensionSubDeclaration3.cpp [hunk 1]
```diff
@@ -135,14 +135,25 @@ void ExtensionSubDeclaration3(gd::ObjectMetadata& obj) {
...
+                _("Change the image of particles (if displayed)."),
+                _("Change the image of particles of _PARAM0_ to _PARAM1_"),
+                _("Advanced"),
+                "CppPlatform/Extensions/particleSystemicon24.png",
+                "CppPlatform/Extensions/particleSystemicon16.png")
+      .AddParameter("object", _("Object"), "ParticleEmitter")
+      .AddParameter("string", _("Image to use"))
+      .SetParameterLongDescription("Indicate the name of the resource");
+
+  obj.AddAction("TextureAmongResources",
```
Comment (D8H) [inline Extensions/ParticleSystem/ExtensionSubDeclaration3.cpp:148]: Actions must now begin with a verb.

If one day a field can do autocompletion as well as expression, one action will need to be deprecated, is it ok?

File: newIDE/app/src/EventsSheet/ParameterFields/ImageResourceField.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import * as React from 'react';
+import ResourceSelector from '../../ResourcesList/ResourceSelector';
+import ResourcesLoader from '../../ResourcesLoader';
+import {
+  type ParameterFieldProps,
+  type ParameterFieldInterface,
+} from './ParameterFieldCommons';
+
+const ImageResourceField = React.forwardRef<
```
Comment (4ian) [inline newIDE/app/src/EventsSheet/ParameterFields/ImageResourceField.js:12]: Any chance you can add it to a story? :)

File: Extensions/ParticleSystem/ExtensionSubDeclaration3.cpp (hunk 1 above)
Comment (4ian) [inline Extensions/ParticleSystem/ExtensionSubDeclaration3.cpp:148]: SetTextureFromResource

(I sadly in the past not used verb at the beginning of actions. But you should have a verb now for actions, Set being the default one)

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/EventsSheet/ParameterFields/ImageResourceField.js (hunk 2 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/ParameterFields/ImageResourceField.js:12]: Yes, will do!

**Pull request merged**